test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,15 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        switch(inputCommand) {
+            case "add": // Fallthrough
+            case "delete": // Fallthrough
+            case "clear": // Fallthrough
+                assertEquals(addressBook, saveFile.load());
```

Comment:
Perhaps another way would be to make use of the `isMutating()` method that you have defined in the Command class?

This would allow you to condense your switch statement chunk thus improving code readability =).
``` java
if(new Parser().parseCommand(inputCommand).isMutating()) {
    assertEquals(addressBook, saveFile.load())
}
```